WcaOnRails/app/models/reassign_wca_id.rb

```diff
@@ -0,0 +1,96 @@
+# frozen_string_literal: true
+
+class ReassignWcaId
+  include ActiveModel::Model
+
+  attr_reader :account1, :account2
+  attr_reader :account1_user, :account2_user
+
+  def account1=(account1)
+    @account1 = account1
+    @account1_user = User.find_by_id(account1)
+  end
+
+  def account2=(account2)
+    @account2 = account2
+    @account2_user = User.find_by_id(account2)
+  end
+
+  validates :account1, presence: true
+  validates :account2, presence: true
+
+  validate :require_valid_accounts
+  def require_valid_accounts
+    if !account1
+      errors.add(:account1, "Not found")
+    end
+    if !account2
+      errors.add(:account2, "Not found")
+    end
+  end
+
+  validate :require_different_people
+  def require_different_people
+    if account1 == account2
+      errors.add(:account2, "Cannot transfer a WCA ID of an account with itself!")
+    end
+  end
+
+  validate :require_valid_wca_ids
+  def require_valid_wca_ids
+    account1_wca_id = @account1_user ? @account1_user.wca_id : nil
+    account2_wca_id = @account2_user ? @account2_user.wca_id : nil
```

Comment:
Nit: you can use the chaining operator like this:

```rb
account2_wca_id = @account2_user&.wca_id
```

Reply:
Awesome! Didn't know about that.